### PR TITLE
treewide: more cleanups of ENABLE_DEBUG

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This is a follow-up PR to clean-up some more `ENABLE_DEBUG`s.

* Few more removals of unneeded `#ENABLE_DEBUG`
* Replace `#if ENABLE_DEBUG`/`if (ENABLE_DEBUG)` with `if (IS_ACTIVE(ENABLE_DEBUG))`
  * This revealed some hidden bugs as well, that have been fixed.

### Testing procedure
Murdock should be happy.

### Issues/PRs references
Continuation of #15274.

### CODE REVIEW

**Comment by basilfx** [comment]
@maribu It seems that the changes in `mtd_spi_nor.c` do compile (w.r.t. your [comment](https://github.com/RIOT-OS/RIOT/pull/15274#discussion_r510688946)) for the waspmote-pro. I don't have that board to test.

**Comment by chrysn** [comment]
There's also a lint in dist/tools/coccinelle/warn/enable_debug_false.cocci that might fit in here well.

**Comment by basilfx** [comment]
> There's also a lint in dist/tools/coccinelle/warn/enable_debug_false.cocci that might fit in here well.

Adapted the rules in a148e5a.

File: sys/net/gnrc/link_layer/lwmac/lwmac.c [hunk 1]
```diff
@@ -923,10 +924,10 @@ static void _lwmac_msg_handler(gnrc_netif_t *netif, msg_t *msg)
...
 #endif
         default: {
-#if ENABLE_DEBUG
-            DEBUG("[LWMAC]: unknown message type 0x%04x"
-                  "(no message handler defined)\n", msg->type);
-#endif
+            if (IS_ACTIVE(ENABLE_DEBUG)) {
+                DEBUG("[LWMAC]: unknown message type 0x%04x"
+                    "(no message handler defined)\n", msg->type);
+            }
```
Comment (maribu) [inline sys/net/gnrc/link_layer/lwmac/lwmac.c:930]:
```suggestion
            DEBUG("[LWMAC]: unknown message type 0x%04x"
                  "(no message handler defined)\n", msg->type);
```

**Review by maribu — COMMENTED** [review]
Looks good to me. Some comments inline

File: cpu/esp32/periph/timer.c [hunk 2]
```diff
@@ -313,13 +313,13 @@ unsigned int IRAM timer_read(tim_t dev)
...
-    DEBUG("%s %u\n", __func__, count_lo);
-    return count_lo;
-    #else
-    return timer_get_counter_lo(dev);
-    #endif
+    if (IS_ACTIVE(ENABLE_DEBUG)) {
+        uint32_t count_lo = timer_get_counter_lo(dev);
+        DEBUG("%s %u\n", __func__, count_lo);
+        return count_lo;
+    } else {
```
[the remark this answers is not in the document]
Comment (basilfx) [reply cpu/esp32/periph/timer.c]: I always mix this with coding conventions from work ;-)

**Comment by basilfx** [comment]
@maribu fixed your comments. Many thanks for the review!

Let me know if I can squash them.

**Comment by maribu** [comment]
> Let me know if I can squash them.

Feel free to squash :-)

**Comment by basilfx** [comment]
Squashed and rebased.

This does need a second ACK for the (minor) core changes though.

File: cpu/esp_common/esp-now/esp_now_netdev.c [hunk 3]
```diff
@@ -154,12 +154,12 @@ static void IRAM_ATTR esp_now_scan_peers_done(void)
...
-    esp_now_peer_num_t peer_num;
-    esp_now_get_peer_num(&peer_num);
-    DEBUG("associated peers total=%d, encrypted=%d\n",
-          peer_num.total_num, peer_num.encrypt_num);
-#endif /* ENABLE_DEBUG */
+    if (IS_ACTIVE(ENABLE_DEBUG)) {
+        esp_now_peer_num_t peer_num;
+        esp_now_get_peer_num(&peer_num);
+        DEBUG("associated peers total=%d, encrypted=%d\n",
+            peer_num.total_num, peer_num.encrypt_num);
```
Comment (maribu) [inline cpu/esp_common/esp-now/esp_now_netdev.c:161]:
```suggestion
        DEBUG("associated peers total=%d, encrypted=%d\n",
              peer_num.total_num, peer_num.encrypt_num);
```

**Review by maribu — APPROVED** [review]
ACK

**Comment by maribu** [comment]
(Feel free to squash the suggested indent change right away.)

**Comment by basilfx** [comment]
Adapted and squashed directly.

**Review by benpicco — APPROVED** [review]
Appears to be mostly mechanical changes, little potential for error.

**Comment by basilfx** [comment]
Thanks for the help!

**Pull request merged**